### ISSUE TITLE
Running benchmarks with bench-compare for one benchmark

### ISSUE THREAD

**Issue by rgommers** [scipy/scipy#4789]
I had expected this to work:

    $ python runtests.py --bench-compare master signal.CalculateWindowedFFT

However, looking at https://github.com/scipy/scipy/blob/master/runtests.py#L198 it's clear that ``bench-compare`` currently only expects commit identifier as arguments. Do we want to improve that, and is that easily doable?

**Comment by pv** [comment]
That works fine for me (cf. extra_argv in runtests; there's also no
benchmark named CalculateWindowedFFT in master). If it doesn't work,
maybe you have some old ASV version?

**Comment by argriffing** [comment]
This syntax has also worked for me, for example in https://github.com/scipy/scipy/pull/4766#issuecomment-95765255. This was with a recent `asv`.

**Comment by argriffing** [comment]
On my system
`$ python runtests.py --bench-compare master signal.CalculateWindowedFFT`
expands to something like
`$ python /path/to/benchmarks/run.py --current-repo continuous -e -f 1.05 b6a114c75e3dd1884b1202b20da19fc98ae44bc6 b6a114c75e3dd1884b1202b20da19fc98ae44bc6 --bench signal.CalculateWindowedFFT`
after `runtests.py` has transformed its command line arguments.

> it's clear that bench-compare currently only expects commit identifier as arguments.

Yes and in your example I think `--bench-compare` sees only `master` not `signal.CalculateWindowedFFT` as an argument, kind of like the following example.

```python
from __future__ import print_function
from argparse import ArgumentParser, REMAINDER
def main():
    parser = ArgumentParser()
    parser.add_argument('--foo', action='append')
    parser.add_argument('more', default=[], nargs=REMAINDER)
    args = parser.parse_args()
    print(args)
main()
```
```
$ python check-cmdline.py --foo a --foo b c --foo d e
Namespace(foo=['a', 'b'], more=['c', '--foo', 'd', 'e'])
```

The benchmarking system is complicated because of the multiple layers of scripts (`runtests.py`, `benchmarks/run.py`, `asv`) and argument transformations.

**Comment by pv** [comment]
If you want to simplify, `benchmarks/run.py` doesn't do anything useful,
except setting CFLAGS so that old versions of Scipy build (except that
they don't, because Cython 0.22 is incompatible with Scipys older than
0.15 or so). Just setting `repo: ".."` in asv.conf.json, removing
run.py, and moving the environment variable stuff to runtests.py would
be quite enough.

**Comment by argriffing** [comment]
> Cython 0.22 is incompatible with Scipys older than 0.15 or so

This seems like a bad situation, if the scipy development version requires Cython 0.22 and recent scipy versions require an older Cython version, especially if tools like `pip` do not support the concept of build-time dependencies (https://www.python.org/dev/peps/pep-0426/). But I don't know what to do about this, and I guess it's getting off-topic for this PR.

I don't know enough about shell-level programming (environment variables, relative paths, etc.) to have any suggestion about how to simplify the benchmarking system. In any case I think the current system is far better than the previous ad-hoc system.

**Comment by rgommers** [comment]
OK thanks, I'll try updating to the latest ``asv`` first.

**Comment by rgommers** [comment]
This should work, so closing.

**Comment by rgommers** [comment]
So Cython 0.22 breaks backwards compat? Was that already reported as a Cython issue and will it be fixed for 0.22.1 or 0.23?

**Comment by argriffing** [comment]
I just checked out `tags/v0.13.3` and ran `$ git clean -dfx` then `runtests.py` and I see
```
Error compiling Cython file:
------------------------------------------------------------
...

    def read(self, n_bytes):
        cdef void *p
        return self.read_string(n_bytes, &p)

    cpdef long int tell(self):
         ^
------------------------------------------------------------

streams.pyx:202:10: Signature not compatible with previous declaration
```
with Cython 0.22.

It's related to https://github.com/scipy/scipy/issues/4137 and https://github.com/scipy/scipy/pull/4084 and the relevant cython PR appears to be https://github.com/cython/cython/commit/e6874284a11fb9cb7d5d5fb7b00c6451812aed20.

I could build `0.14.1` with Cython 0.22.
